src/psyclone/psyir/nodes/node.py

```diff
@@ -1629,6 +1629,31 @@ def debug_string(self):
         from psyclone.psyir.backend.debug_writer import DebugWriter
         return DebugWriter()(self)
 
+    def origin_string(self):
+        ''' Generates a string with the available information about where
+        this node has been created. It currently only works with Fortran
+        Statements or subchildren of them.
+
+        :returns: a string specifing the origin of this node.
+        :rtype: str
+        '''
+        name = self.coloured_name(False)
+        line_span = "<unknown>"
+        original_src = "<unknown>"
+        filename = "<unknown>"
+        from psyclone.psyir.nodes.statement import Statement
+        node = self if isinstance(self, Statement) else self.ancestor(Statement)
```

Comment:
You can just do `node=self.ancestor(Statement, include_self=True)`

Reply:
Done